MinIO Erasure Stripe Size (#1183)
minio/docs: source/reference/minio-server/settings/core.rst

   Some	types of files cannot be significantly reduced in size.
   MinIO will *not* compress these, even if specified in an :mc-conf:`~compression.mime_types` argument.
   See :ref:`Excluded types <minio-data-compression-excluded-types>` for details.

Erasure Stripe Size
~~~~~~~~~~~~~~~~~~~

.. tab-set:: 

   .. tab-item:: Environment Variable
      :sync: envvar

      .. envvar:: MINIO_ERASURE_SET_DRIVE_COUNT

   .. tab-item:: Configuration Variable
      :sync: config

      .. include:: /includes/common-mc-admin-config.rst
         :start-after: start-minio-settings-no-config-option
         :end-before: end-minio-settings-no-config-option

*Optional*

The :ref:`erasure set size <minio-ec-basics>` to apply for all drives in a given :term:`server pool`.

> You **must** set this value before you initialize the cluster.

Reviewer: Saying `must` here just below the `optional` is odd.

Perhaps
```suggestion
If you change to this value, you **must** do so *before* you initialize the cluster.
```

Author: ```suggestion
If you set this value, you **must** do so *before* you initialize the cluster
```

Slight tweak since it's not a 'change' per-se, and we don't support 'changing' it later. But yes.

   Some	types of files cannot be significantly reduced in size.
   MinIO will *not* compress these, even if specified in an :mc-conf:`~compression.mime_types` argument.
   See :ref:`Excluded types <minio-data-compression-excluded-types>` for details.

Erasure Stripe Size
~~~~~~~~~~~~~~~~~~~

.. tab-set:: 

   .. tab-item:: Environment Variable
      :sync: envvar

      .. envvar:: MINIO_ERASURE_SET_DRIVE_COUNT

   .. tab-item:: Configuration Variable
      :sync: config

      .. include:: /includes/common-mc-admin-config.rst
         :start-after: start-minio-settings-no-config-option
         :end-before: end-minio-settings-no-config-option

*Optional*

The :ref:`erasure set size <minio-ec-basics>` to apply for all drives in a given :term:`server pool`.

You **must** set this value before you first start the server pool nodes.
The stripe size selected for the first server pool becomes a baseline minimum for any additional server pools.

> You **cannot** change the erasure set stripe size once the deployment starts.

Reviewer 2: ```suggestion
You **cannot** change the erasure set stripe size after the cluster has been initialized.
```

**Do not** deploy custom stripe sizes into production without first validating in lower environments using a mix of performance, stress, and real-workload tests.

.. important::

   Stripe size impacts multiple areas of functionality, including:

   - Storage Capacity
   - Failure Tolerance
   - Parity Configuration
   - Workload Efficiency

   MinIO provides the `Erasure Code Calculator <https://min.io/product/erasure-code-calculator?jmp=docs>`__ to provide a baseline of potential configurations for your deployments.

>    |subnet| users should log in and open an issue to discuss stripe size settings prior to implementing them in testing or production workloads.

Reviewer 3: This phrase is reading odd for me. Perhaps something like one of these?
- prior to implementing them for test or production workloads
- prior to implementing them in test or production

Reviewer: ```suggestion
   |subnet| users should log in and open an issue to discuss stripe size settings prior to implementing them in test or production environments.
```

Reviewer: > any environment

Including their local PC for development purposes? I think that's maybe too broad.

Author: :thinking: I'm inclined to keep it. It is kind of broad, but I am certain the single-pc local dev is going to ignore what we say anyways.

>    MinIO provides the `Erasure Code Calculator <https://min.io/product/erasure-code-calculator?jmp=docs>`__ to provide a baseline of potential configurations for your deployments.
>    |subnet| users should log in and open an issue to discuss stripe size settings prior to implementing them in testing or production workloads.
> 
>    MinIO's default stripe selection algorithms are typically sufficient for the majority of workloads.

Author: ```suggestion
   MinIO's default stripe selection algorithms are typically sufficient for the majority of workloads.

   **Do not** change the stripe size setting unless directed to by MinIO engineering.   
   |subnet| users should log in and open an issue to discuss stripe size settings prior to implementing them in testing or production workloads.

   
```

@klauspost perhaps a good mid-point here.

Erasure Stripe Size
~~~~~~~~~~~~~~~~~~~

.. tab-set:: 

   .. tab-item:: Environment Variable
      :sync: envvar

      .. envvar:: MINIO_ERASURE_SET_DRIVE_COUNT

   .. tab-item:: Configuration Variable
      :sync: config

      .. include:: /includes/common-mc-admin-config.rst
         :start-after: start-minio-settings-no-config-option
         :end-before: end-minio-settings-no-config-option

*Optional*

The :ref:`erasure set size <minio-ec-basics>` to apply for all drives in a given :term:`server pool`.

You **must** set this value before you first start the server pool nodes.
The stripe size selected for the first server pool becomes a baseline minimum for any additional server pools.

You **cannot** change the erasure set stripe size after the cluster has been initialized.

**Do not** deploy custom stripe sizes into production without first validating in lower environments using a mix of performance, stress, and real-workload tests.

.. important::

   Stripe size impacts multiple areas of functionality, including:

   - Storage Capacity
   - Failure Tolerance
   - Parity Configuration
   - Workload Efficiency

>    MinIO's default stripe selection algorithms are typically sufficient for the majority of workloads.

Reviewer: ```suggestion
   MinIO's stripe selection algorithms set appropriate defaults for the majority of workloads.
   Changing the stripe size from this default is unusual and generally not necessary or advised.
```

Reviewer: Maybe?

Reviewer 4: I like this, it's more assertive, much less wiggle room for people to say "well, I'm special, not typical".  Much harder for someone to believe, "well, I'm inappropriate, not appropriate."
